apps/test/unit/templates/teacherNavigation/TeacherNavigationBarTest.jsx

```diff
@@ -33,41 +51,85 @@ describe('TeacherNavigationBar', () => {
 
   let store;
 
-  beforeEach(() => {
+  const renderDefault = (selectedSectionId = 11) => {
     store = getStore();
     registerReducers({
       teacherSections,
     });
     store.dispatch(setSections(serverSections));
-  });
 
-  test('renders correctly with visible sections', () => {
     render(
       <Provider store={store}>
-        <MemoryRouter>
-          <TeacherNavigationBar />
-        </MemoryRouter>
+        <RouterProvider
+          router={createMemoryRouter(
+            createRoutesFromElements([
+              <Route
+                path={TEACHER_NAVIGATION_SECTIONS_URL}
+                element={
+                  <div>
+                    <TeacherNavigationBar />
+                    <Outlet />
+                  </div>
+                }
+              >
+                <Route
+                  path={SPECIFIC_SECTION_BASE_URL}
+                  element={
+                    <div>
+                      <Outlet />
+                    </div>
+                  }
+                  loader={({params}) => {
+                    store.dispatch(selectSection(params.sectionId));
+                    return null;
+                  }}
+                >
+                  <Route
+                    path={'progress'}
+                    element={
+                      <div>
+                        <LocationElement location={location} />
+                      </div>
+                    }
+                  />
+                  <Route
+                    path={'manage_students'}
+                    element={
+                      <div>
+                        <LocationElement location={location} />
+                      </div>
+                    }
+                  />
+                </Route>
+              </Route>,
+            ]),
+            {
+              initialEntries: [
+                `/teacher_dashboard/sections/${selectedSectionId}/progress`,
+              ],
+              basename: TEACHER_NAVIGATION_BASE_URL,
+            }
+          )}
+        />
       </Provider>
     );
+  };
+
+  test('renders correctly with visible sections', async () => {
+    renderDefault();
 
-    screen.getByText(i18n.classSections());
+    await screen.findByText(i18n.classSections());
     screen.getByRole('combobox');
-    screen.getByText('Period 1');
+    expect(screen.getByText('Period 1')).toBeVisible();
```

Comment:
Can I ask why you added the `.toBeVisible` line here?

Reply:
Very slight improvement, this tests that the `period 1` is visible and selected in the dropdown. The other options should be invisible because the dropdown is not open.